### PR TITLE
Fix watch mode of Karma tests

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: #3403 

This PR does two things:

1) Speeds up compiling since we now only create one bundle.
1) Fixes the Karma watch command so tests can be run in the background while coding.

**Test Plan**
- [ ] Run `npm run test:karma:watch` make sure only one test bundle is compiled, and that it reruns every time you save a file.

### CODE REVIEW

File: client/karma.conf.js [hunk 1]
```diff
@@ -17,15 +15,15 @@ module.exports = function(config) {
...
+      'test/karma/test-index.js'
     ],
 
     mochaReporter: {
       showDiff: true
     },
 
     preprocessors: {
-      [karmaTestPattern]: ['webpack', 'sourcemap']
+      'test/karma/test-index.js': ['webpack']
```
Comment (joofsh) [inline client/karma.conf.js]: this disables sourcemaps?

File: client/karma.conf.js [hunk 2]
```diff
@@ -35,6 +33,7 @@ module.exports = function(config) {
     // for shims and polyfills, need to be manually imported
```
Comment (joofsh) [inline client/karma.conf.js]: Is this comment out of date now? the "each file" part?

File: client/karma.conf.js (hunk 1 above)
Comment (joofsh) [reply client/karma.conf.js]: Okay I tested it a bunch and when there's a compile error, it's not super clear what file it happened in. Fortunately, re-enabling the `sourcemap` option here doesn't seem to help so....strictly better 👍

Comment (mdbenjam) [reply client/karma.conf.js]: I think I read some about source maps. I'll add that to things to investigate in this ticket.

File: client/karma.conf.js (hunk 2 above)
Comment (mdbenjam) [reply client/karma.conf.js]: I think it is. Removed.

**Pull request merged**